### PR TITLE
[Doc] add 3.0.9 rn

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Why I'm doing:

What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/release_notes/release-3.0.md [hunk 1]
```diff
@@ -4,6 +4,90 @@ displayed_sidebar: "English"
...
+- Added session variable `enable_materialized_view_for_insert`, which controls whether materialized views rewrite the queries in INSERT INTO SELECT statements. The default value is `false`. [#37505](https://github.com/StarRocks/starrocks/pull/37505)
+- Added session variables `transaction_read_only` and `tx_read_only` to specify the transaction access mode, which are compatible with MySQL versions 5.7.20 and above. [#37249](https://github.com/StarRocks/starrocks/pull/37249)
+
+#### FE configurations
+
+- Change the FE configuration item `enable_new_publish_mechanism` to a static parameter from a dynamic one. You must restart the FE after you modify the parameter settings. [#35338](https://github.com/StarRocks/starrocks/pull/35338)
+- Added the [FE configuration item](https://docs.starrocks.io/zh/docs/administration/Configuration/#配置-fe-动态参数) `routine_load_unstable_threshold_second` is added. [#36222](https://github.com/StarRocks/starrocks/pull/36222)
```
Comment (evelynzhaojie) [inline docs/en/release_notes/release-3.0.md:50]: the link is chinese

File: docs/en/release_notes/release-3.0.md [hunk 2]
```diff
@@ -4,6 +4,90 @@ displayed_sidebar: "English"
...
+- Added the BE configuration item `pindex_major_compaction_limit_per_disk` to configure the maximum concurrency of compaction on a disk. This addresses the issue of uneven I/O across disks due to compaction. This issue can cause excessively high I/O for certain disks. The default value is `2`. [#36681](https://github.com/StarRocks/starrocks/pull/36681)
+- Added BE configuration items to specify the timeout duration for connecting to object storage:
+  - `object_storage_connect_timeout_ms`: Timeout duration to establish socket connections with object storage. The default value is `-1`, which means to use the default timeout duration of the SDK configurations.
+  - `object_storage_request_timeout_ms`:  Timeout duration to establish HTTP connections with object storage. The default value is `-1`, which means to use the default timeout duration of the SDK configurations.
+
+#### Bug Fixes
+
+Fixed the following issues:
+
+- In some cases, BEs may crash when a Catalog is used to read ORC external tables.
```
Comment (evelynzhaojie) [inline docs/en/release_notes/release-3.0.md:66]: empty lines in 67 and 69

File: docs/en/release_notes/release-3.0.md [hunk 3]
```diff
@@ -4,6 +4,90 @@ displayed_sidebar: "English"
...
+Release date: January 2, 2024
+
+## New features
+
+- Added the [percentile_disc](../sql-reference/sql-functions/aggregate-functions/percentile_disc.md) function. [#36352](https://github.com/StarRocks/starrocks/pull/36352)
+- Added a new metric `max_tablet_rowset_num` for setting the maximum allowed number of rowsets. This metric helps detect possible compaction issues and thus reduces the occurrences of the error "too many versions". [#36539](https://github.com/StarRocks/starrocks/pull/36539)
+
+## Improvements
+
+- A new value option `GROUP_CONCAT_LEGACY` is added to the session variable [sql_mode](https://docs.starrocks.io/zh/docs/reference/System_variable/#sql_mode) to provide compatibility with the implementation logic of the [group_concat](https://docs.starrocks.io/zh/docs/sql-reference/sql-functions/string-functions/group_concat/) function in versions earlier than v2.5. [#36150](https://github.com/StarRocks/starrocks/pull/36150)
```
Comment (evelynzhaojie) [inline docs/en/release_notes/release-3.0.md:18]: two zh links

File: docs/en/release_notes/release-3.0.md [hunk 4]
```diff
@@ -4,6 +4,90 @@ displayed_sidebar: "English"
...
+- When using JDK, the default GC algorithm is G1GC. [#37386](https://github.com/StarRocks/starrocks/pull/37386)
+- The `be_tablets` view in the `information_schema` database provides a new field `INDEX_DISK`, which records the disk usage (measured in bytes) of persistent indexes [#35615](https://github.com/StarRocks/starrocks/pull/35615)
+- Queries on MySQL external tables and the external tables within JDBC catalogs support including keywords in the WHERE clause. [#35917](https://github.com/StarRocks/starrocks/pull/35917)
+- Supports updates onto the specified partitions of an automatically partitioned table. If the specified partitions do not exist, an error is returned. [#34777](https://github.com/StarRocks/starrocks/pull/34777)
+- The Primary Key table size returned by the [SHOW DATA](https://docs.starrocks.io/zh/docs/sql-reference/sql-statements/data-manipulation/SHOW_DATA/) statement includes the sizes of **.cols** files (these are files related to partial column updates and generated columns) and persistent index files. [#34898](https://github.com/StarRocks/starrocks/pull/34898)
```
Comment (evelynzhaojie) [inline docs/en/release_notes/release-3.0.md:23]: zh link

File: docs/en/release_notes/release-3.0.md [hunk 5]
```diff
@@ -4,6 +4,90 @@ displayed_sidebar: "English"
...
+- Optimized the performance of persistent index update when compaction is performed on all rowsets of a Primary Key table, which reduces disk read I/O. [#36819](https://github.com/StarRocks/starrocks/pull/36819)
+- When the string on the right side of the LIKE operator within the WHERE clause does not include `%` or `_`, the LIKE operator is converted into the `=` operator. [#37515](https://github.com/StarRocks/starrocks/pull/37515)
+- Optimized the logic used to compute compaction scores for Primary Key tables, thereby aligning the compaction scores for Primary Key tables within a more consistent range with the other three table types. [#36534](https://github.com/StarRocks/starrocks/pull/36534)
+- The result returned by the [SHOW ROUTINE LOAD](https://docs.starrocks.io/zh/docs/sql-reference/sql-statements/data-manipulation/SHOW_ROUTINE_LOAD/) statement now includes the timestamps of consumption messages from each partition. [#36222](https://github.com/StarRocks/starrocks/pull/36222)
```
Comment (evelynzhaojie) [inline docs/en/release_notes/release-3.0.md:27]: zh link

File: docs/zh/release_notes/release-3.0.md [hunk 6]
```diff
@@ -4,6 +4,88 @@ displayed_sidebar: "Chinese"
...
 
+# 3.0.9
+
+发布日期：2024 年 1 月 2 日
+
+## 新增特性
+
+- 新增聚合函数 [percentile_disc](../sql-reference/sql-functions/aggregate-functions/percentile_disc.md)。[#36352](https://github.com/StarRocks/starrocks/pull/36352)
+- 新增监控指标 `max_tablet_rowset_num`（用于设置 Rowset 的最大数量），可以协助提前发现 Compaction 是否会出问题并及时干预，减少报错信息“too many versions”的出现。[#36539](https://github.com/StarRocks/starrocks/pull/36539)
+- 提供了IO profiler工具用于定位IO高的tablet或者操作类型。 https://github.com/StarRocks/starrocks/pull/36320  [问题排查常用脚本](https://starrocks.feishu.cn/docx/FbOvdZjeJoedzmxMFUQcugFRnXb)
```
Comment (evelynzhaojie) [inline docs/zh/release_notes/release-3.0.md:15]: 毛欣在文档里写了，这个不放 到 release notes 里，只做内部记录

File: docs/zh/release_notes/release-3.0.md [hunk 7]
```diff
@@ -4,6 +4,88 @@ displayed_sidebar: "Chinese"
...
+
+- 系统变量 [sql_mode](https://docs.starrocks.io/zh/docs/reference/System_variable/#sql_mode) 增加 `GROUP_CONCAT_LEGACY` 选项，用以兼容 [group_concat](https://docs.starrocks.io/zh/docs/sql-reference/sql-functions/string-functions/group_concat/) 函数在 2.5（不含）版本之前的实现逻辑。[#36150](https://github.com/StarRocks/starrocks/pull/36150)
+- 使用 JDK 时 GC 算法默认采用 G1GC。[#37386](https://github.com/StarRocks/starrocks/pull/37386)
+- 在 `be_tablets` 表中增加 `INDEX_DISK` 记录持久化索引的磁盘使用量，单位是 Bytes。[#35615](https://github.com/StarRocks/starrocks/pull/35615)
+- 支持 MySQL 外部表和 JDBC Catalog 外部表的 WHERE 子句中包含关键字。[#35917](https://github.com/StarRocks/starrocks/pull/35917)
+- 如果是自动分区表，也支持指定分区名进行更新，如果分区不存在则报错。[#34777](https://github.com/StarRocks/starrocks/pull/34777)
+- 主键模型表 [SHOW DATA](https://docs.starrocks.io/zh/docs/sql-reference/sql-statements/data-manipulation/SHOW_DATA/) 的结果中新增包括 **.cols** 文件（部分列更新和生成列相关的文件）和持久化索引文件的文件大小信息。[#34898](https://github.com/StarRocks/starrocks/pull/34898)
+- 优化主键模型表全部 Rowset 进行 Compaction 时的持久化索引更新性能，降低 I/O 负载。 [#36819](https://github.com/StarRocks/starrocks/pull/36819)
+- WHERE 子句中 LIKE 运算符右侧字符串中不包括 `%` 或者 `_` 时，LIKE 运算符会转换成 = 运算符。[#37515](https://github.com/StarRocks/starrocks/pull/37515)
```
Comment (evelynzhaojie) [inline docs/zh/release_notes/release-3.0.md:26]: = 加 codeph 格式

File: docs/zh/release_notes/release-3.0.md [hunk 8]
```diff
@@ -4,6 +4,88 @@ displayed_sidebar: "Chinese"
...
+- 2.4 及以下的版本升级到高版本，可能会出现 Compaction Score 很高的问题。[#34618](https://github.com/StarRocks/starrocks/pull/34618)
+- 使用 MariaDB ODBC Driver 查询 `INFORMATION_SCHEMA` 中的信息时，`schemata` 视图中 `CATALOG_NAME` 列中取值都显示的是 `null`。[#34627](https://github.com/StarRocks/starrocks/pull/34627)
+- 导入数据异常导致 FE Crash 后无法重启。[#34590](https://github.com/StarRocks/starrocks/pull/34590)
+- Stream Load 导入作业在 **PREPARD** 状态下，同时有 Schema Change 在执行，会导致数据丢失。[#34381](https://github.com/StarRocks/starrocks/pull/34381)
+- 如果 HDFS 路径以两个或以上斜杠（`/`）结尾，HDFS 备份恢复会失败。[#34601](https://github.com/StarRocks/starrocks/pull/34601)
+- 使用 ALTER TABLE 增加 `partition_live_number` 属性没有生效。[#34842](https://github.com/StarRocks/starrocks/pull/34842)
+- ARRAY_DISTINCT 函数偶发 BE Crash。[#36377](https://github.com/StarRocks/starrocks/pull/36377)
+- 某些情况下，物化视图刷新可能会出现死锁问题。[#35736](https://github.com/StarRocks/starrocks/pull/35736)
+- 特定场景下 Global Runtime Filter 可能会引发 BE crash。[#35776](https://github.com/StarRocks/starrocks/pull/35776)
+- 某些情况下 `bitmap_to_string` 会因为内存溢出导致查询结果错误。[#37405](https://github.com/StarRocks/starrocks/pull/37405)
```
Comment (evelynzhaojie) [inline docs/zh/release_notes/release-3.0.md:87]: 最后这条 Bug 中文改一下，和飞书一样

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Pull request merged**